logger/logger.go

```diff
@@ -188,33 +215,41 @@ func (logger *Logger) Critical(message string) {
 
 // LogWithOptions allows the user to update options uniquely for a given log message
 // before sending the log to LogDNA.
-func (logger *Logger) LogWithOptions(message string, options Options) {
-	checkOptions(&options)
+func (logger *Logger) LogWithOptions(message string, options Options) error {
+	err := options.validate()
+	if err != nil {
+		return err
+	}
 
 	loggerMessage := Message{
 		Body:    message,
 		Options: options,
 	}
 
 	logger.Messages <- loggerMessage
+	return nil
 }
 
 // LogWithLevelAndApp allows the user to customize level and app uniquely for a single log
 // before sending the log into LogDNA.
-func (logger *Logger) LogWithLevelAndApp(message string, level string, app string) {
-	checkParameterLength("Level", level)
-	checkParameterLength("App", app)
+func (logger *Logger) LogWithLevelAndApp(message string, level string, app string) error {
```

Comment:
Also, I'm not sure if this a normal thing in Go, but why don't we just have `Log` and `LogWithOptions`? Both of those should be able to handle the possibilities of `log w/ level`, `log w/ app`, `log w/ app and level` plus the other countless combinations.

Probably outside the scope of this PR, but wanted to bring it up.

Reply:
agreed, I actually reworked this in an upcoming branch already so that `LogWithLevel` and `LogWithLevelAndApp` are just calling `LogWithOptions` anyway. I am onboard with removing these and will do that alongside the refactoring and tests in LOG-6966